functions/Remove-DbaDbUser.ps1

```diff
@@ -0,0 +1,230 @@
+        [parameter(ParameterSetName='Object')]
+        [switch]$Force,
+        
+		[switch][Alias('Silent')]$EnableException
+    )
+
+    begin {
+		function Remove-DbUser {
+			[CmdletBinding()]
+			param ([Microsoft.SqlServer.Management.Smo.User[]]$users)
+
+            foreach ($user in $users) {
+                $db = $user.Parent
+                $server = $db.Parent
+                Write-Message -Level Verbose -Message "Removing User $user from Database $db on target $server"
+
+                # Drop Schemas owned by the user before droping the user
+                $schemaUrns = $user.EnumOwnedObjects() | Where-Object Type -EQ Schema
+                if ($schemaUrns) {
+                    Write-Message -Level Verbose -Message "User $user owns $($schemaUrns.Count) schema(s)."
+                    
+                    # Need to gather up the schema changes so they can be done in a non-desctructive order
+                    $alterSchemas = @()
+                    $dropSchemas = @()
+
+                    foreach ($schemaUrn in $schemaUrns) {
+                        $schema = $server.GetSmoObject($schemaUrn)
+
+                        # Drop any schema that is the same name as the user
+                        if ($schema.Name -EQ $user.Name) {
+                            # Check for owned objects early so we can exit before any changes are made
+                            $ownedUrns = $schema.EnumOwnedObjects()
+                            if (-Not $ownedUrns) {
+                                $dropSchemas += $schema
+                            } else {
+                                Write-Message -Level Warning -Message "User owns objects in the database and will not be removed."
+                                foreach ($ownedUrn in $ownedUrns) {
+                                    $obj = $server.GetSmoObject($ownedUrn)
+                                    Write-Message -Level Warning -Message "User $user owns $($obj.GetType().Name) $obj"
+                                }
+                                return  # exit function
+                            }
+                        }
+
+                        # Change the owner of any schema not the same name as the user
+                        if ($schema.Name -NE $user.Name) {
+                            # Check for owned objects early so we can exit before any changes are made
+                            $ownedUrns = $schema.EnumOwnedObjects()
+                            if (($ownedUrns -And $Force) -Or (-Not $ownedUrns)) {
+                                $alterSchemas += $schema
+                            } else {
+                                Write-Message -Level Warning -Message "User $user owns the Schema $schema, which owns $($ownedUrns.Count) Object(s).  If you want to change the schemas' owner to [dbo] and drop the user anyway, use -Force parameter.  User $user will not be removed."
+                                return  #exit the function
```

Comment:
I would believe this should be a Stop-Function call with a continue and not a return. If more than one schema is passed in wouldn't this exit the function even if there were additional schemas in the collection?

Reply:
Yes, it would exit the function.  My thinking was to do as little change as possible if it wasn't going to be able to drop the user anyway.  Stop-Function is new to me, so if that's the standard, can certainly switch to it.

Reply:
OK, I see there was still an issue in that it would exit even if it had more than one user to go through.  I refactored the code, see PR 2647: https://github.com/sqlcollaborative/dbatools/pull/2647